src/adapter/etl-adapter-google-sheet/src/Flow/ETL/Adapter/GoogleSheet/GoogleSheetLoader.php

```diff
@@ -0,0 +1,179 @@
+
+        $normalizer = new RowsNormalizer(
+            new EntryNormalizer($context->config->caster(), $this->dateTimeFormat, $this->dateFormat)
+        );
+
+        $this->service->spreadsheets_values->update(
+            $this->spreadsheetId,
+            $sheetRange->toString(),
+            new ValueRange(
+                [
+                    'values' => $this->values($rows, $normalizer),
+                    'majorDimension' => 'ROWS',
+                ]
+            ),
+            $this->inputOption->toArray()
+        );
+
+        $this->loadedRows = $currentRow;
+    }
+
+    public function withDateFormat(string $dateFormat) : self
+    {
+        $this->dateFormat = $dateFormat;
+
+        return $this;
+    }
+
+    public function withDateTimeFormat(string $dateTimeFormat) : self
+    {
+        $this->dateTimeFormat = $dateTimeFormat;
+
+        return $this;
+    }
+
+    public function withHeader(bool $withHeader) : self
+    {
+        $this->withHeader = $withHeader;
+
+        return $this;
+    }
+
+    public function withInputOption(ValueInputOption $inputOption) : self
+    {
+        $this->inputOption = $inputOption;
+
+        return $this;
+    }
+
+    private function calculateCurrentRow(Rows $rows) : int
+    {
+        return $rows->count() + (0 === $this->loadedRows ? ($this->withHeader ? 1 : 0) : $this->loadedRows);
+    }
+
+    private function calculateSheetAddress(int $columnsCount, int $currentRow) : SheetAddress
+    {
+        return SheetAddress::calculate(
+            $columnsCount,
+            $currentRow,
+            new SheetAddress(
+                startCell: 0 === $this->loadedRows ? 'A1' : 'A' . $this->loadedRows + 1,
+                sheetName: $this->sheetName
+            )
+        );
+    }
+
+    private function increaseSheetSize(int $currentRow, int $columnsCount) : void
+    {
+        // Check if we need to increase sheet size from default 5k rows
+        if (!$this->isDividable($currentRow) && !$this->isDividable($currentRow - 1)) {
+            return;
+        }
+
+        $this->service->spreadsheets->batchUpdate(
+            $this->spreadsheetId,
+            new BatchUpdateSpreadsheetRequest([
+                'requests' => [
+                    [
+                        'updateSheetProperties' => [
+                            'properties' => [
+                                'sheetId' => $this->sheetId(),
+                                'gridProperties' => [
+                                    'rowCount' => $currentRow + 5_000,
+                                    // Ensure we update columns size as well
+                                    'columnCount' => $columnsCount,
+                                ],
+                            ],
+                            'fields' => 'gridProperties(rowCount,columnCount)',
+                        ],
+                    ],
+                ],
+            ])
+        );
+    }
+
```

Comment:
I'm not suer if that's the correct logic here, here is how I would do the dynamic rows allocation. 

1) when load is called for the first time get the current size of spreadsheet 
2) before loading rows check if we are not going to exceed the size of spreadsheet
  a) if spreadsheet is not large enough - expand it but 1k - but allow to modify this through `withSheetDynamicAllocationSize()` 
  b) if there is enough space just keep loading 
  
Additionally with each load you should also check if number of entires is smaller than number of allocated columns, if it's not, you should also add columns.